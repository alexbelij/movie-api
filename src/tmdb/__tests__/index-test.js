/* @flow */

import {modelFromObject} from '../../testUtils';
import Tmdb from '../index';

const movieId = 140607; // Star Wars: The Force Awakens
const tvShowId = 1399; // Game of Thrones

describe('TMDB', () => {
  let tmdb: Tmdb;

  beforeAll(() => {
    tmdb = new Tmdb();
  });

  it('fetches movie id by imdb id', async () => {
    expect(await tmdb.getMovieId({imdbId: 'tt2488496'})).toMatchSnapshot();
  });

  it('fetches movie id by title', async () => {
    expect(
      await tmdb.getMovieId({
        title: 'Star Wars: The Force Awakens',
      }),
    ).toMatchSnapshot();
  });

  it('fetches tv show id by imdb id', async () => {
    expect(
      await tmdb.getTvShowId({
        imdbId: 'tt0944947',
      }),
    ).toMatchSnapshot();
  });

  it('fetches tv show id by title', async () => {
    expect(
      await tmdb.getTvShowId({
        title: 'Game of Thrones',
      }),
    ).toMatchSnapshot();
  });

  it('fetches movie info from tmdb for a given id', async () => {
    expect(modelFromObject(await tmdb.getMovieInfo(movieId))).toMatchSnapshot();
  });

  it('fetches tv show info from tmdb for a given id', async () => {
    expect(
      modelFromObject(await tmdb.getTvShowInfo(tvShowId)),
    ).toMatchSnapshot();
  });
});
