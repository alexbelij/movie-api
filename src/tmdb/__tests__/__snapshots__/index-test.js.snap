// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TMDB fetches movie id by imdb id 1`] = `140607`;

exports[`TMDB fetches movie id by title 1`] = `140607`;

exports[`TMDB fetches movie info from tmdb for a given id 1`] = `
Object {
  "backdropUrl": "string",
  "budget": "number",
  "credits": Object {
    "cast": Array [
      Object {
        "character": "string",
        "name": "string",
        "photoUrl": "string",
      },
    ],
    "crew": Object {
      "cinematographers": Array [
        Object {
          "department": "string",
          "job": "string",
          "name": "string",
          "photoUrl": "null",
        },
      ],
      "composers": Array [
        Object {
          "department": "string",
          "job": "string",
          "name": "string",
          "photoUrl": "string",
        },
      ],
      "directors": Array [
        Object {
          "department": "string",
          "job": "string",
          "name": "string",
          "photoUrl": "string",
        },
      ],
      "producers": Array [
        Object {
          "department": "string",
          "job": "string",
          "name": "string",
          "photoUrl": "string",
        },
      ],
      "writers": Array [
        Object {
          "department": "string",
          "job": "string",
          "name": "string",
          "photoUrl": "string",
        },
      ],
    },
  },
  "genres": "Array<string>",
  "homepage": "string",
  "imdbId": "string",
  "keywords": "Array<string>",
  "originalLanguage": "string",
  "originalTitle": "string",
  "posterUrl": "string",
  "productionCompanies": "Array<string>",
  "productionCountries": Array [
    Object {
      "iso_3166_1": "string",
      "name": "string",
    },
  ],
  "releaseDate": "string",
  "revenue": "number",
  "runtime": "number",
  "synopsis": "string",
  "tagline": "string",
  "title": "string",
  "tmdbId": "number",
  "tmdbPopularity": "number",
  "tmdbRating": "number",
  "tmdbRatingVoteCount": "number",
  "videos": Array [
    Object {
      "iso_3166_1": "string",
      "iso_639_1": "string",
      "key": "string",
      "name": "string",
      "site": "string",
      "size": "number",
      "type": "string",
    },
  ],
}
`;

exports[`TMDB fetches tv show id by imdb id 1`] = `1399`;

exports[`TMDB fetches tv show id by title 1`] = `1399`;

exports[`TMDB fetches tv show info from tmdb for a given id 1`] = `
Object {
  "backdropUrl": "string",
  "createdBy": Array [
    Object {
      "name": "string",
      "photoUrl": "string",
    },
  ],
  "credits": Object {
    "cast": Array [
      Object {
        "character": "string",
        "name": "string",
        "photoUrl": "string",
      },
    ],
    "crew": Array [
      Object {
        "department": "string",
        "job": "string",
        "name": "string",
        "photoUrl": "string",
      },
    ],
  },
  "episodeRuntime": "Array<number>",
  "firstAirDate": "string",
  "genres": "Array<string>",
  "homepage": "string",
  "inProduction": "boolean",
  "keywords": "Array<string>",
  "languages": "Array<string>",
  "lastAirDate": "string",
  "name": "string",
  "networks": "Array<string>",
  "numberOfEpisodes": "number",
  "numberOfSeasons": "number",
  "originCountry": "Array<string>",
  "originalLanguage": "string",
  "originalName": "string",
  "posterUrl": "string",
  "productionCompanies": "Array<string>",
  "seasons": Array [
    Object {
      "airDate": "string",
      "episodeCount": "number",
      "posterUrl": "string",
      "seasonNumber": "number",
    },
  ],
  "status": "string",
  "synopsis": "string",
  "tmdbId": "number",
  "tmdbPopularity": "number",
  "tmdbRating": "number",
  "tmdbRatingVoteCount": "number",
  "type": "string",
  "videos": Array [],
}
`;
